### ISSUE TITLE
}bedrock.cube.view.create pSubN = 1 add N level even the dimension specified in sFilter 

### ISSUE THREAD

**Issue by cw-cleung** [cubewise-code/bedrock#251]
**Describe the bug**
}bedrock.cube.view.create pSubN = 1 add N level even the dimension specified in sFilter, if the dimension specified has different case then the principal dim name, e.g. "M Inventory In Shop View" vs "M Inventory in Shop View". 

**To Reproduce**
Dimension name is "M Inventory in Shop View", but in sFilter it is specified as "M Inventory In Shop View". 
It successfully assigned "Total Inventory" (C level) to the view first, but in bedrock `if pSubN = 1` (line 389), it again assign all N level, as it compare by Scan function (line 406), which is case sensitive. 
**Maybe better to compare the Upper string, to avoid case sensitivity issue.** 

```
sFilter = '';
sFilter = sFilter | 'Scenario' | sDelimEleStart | cScenario;
sFilter = sFilter | sDelimDim | 'Location Set' | sDelimEleStart | 'All Location Sets';
sFilter = sFilter | sDelimDim | 'Data Source' | sDelimEleStart | 'All Data Sources';
sFilter = sFilter | sDelimDim | 'M Inventory In Shop View' | sDelimEleStart | 'Total Inventory';

ExecuteProcess('}bedrock.cube.view.create',
   'pLogOutput', 0,
   'pCube', cCubeSource,
   'pView', cView,
   'pFilter', sFilter,
   'pSuppressZero', 1,
   'pSuppressConsol', 0,
   'pSuppressRules', 0,
   'pSuppressConsolStrings', 0,
   'pDimDelim', sDelimDim,
   'pEleStartDelim', sDelimEleStart,
   'pEleDelim', sDelimEle,
   'pTemp', 0,
   'pSubN', 1
  );
```

**Expected behavior**
All specified dimension in sFilter should keep untouched even pSubN = 1.

**Comment by lotsaram** [comment]
Hi @MateuszOszer as you are doing some other bedrock work could you also handle this one please?  
It seems pretty simple to fix. When the Scan comparison is being done on dimension names in the filter vs. in the cube just make sure to wrap **_BOTH_** in either Lower() or Upper() to avoid any issues as described.

**Comment by lotsaram** [comment]
Fixed in #250
